src/com/google/enterprise/adaptor/database/UniqueKey.java

```diff
@@ -82,27 +85,26 @@
       def[1] = def[1].trim();
       String name = def[0];
       ColumnType type;
-      if ("int".equals(def[1].toLowerCase())) {
-        type = ColumnType.INT;
-      } else if ("string".equals(def[1].toLowerCase())) {
-        type = ColumnType.STRING;
-      } else if ("timestamp".equals(def[1].toLowerCase())) {
-        type = ColumnType.TIMESTAMP;
-      } else if ("date".equals(def[1].toLowerCase())) {
-        type = ColumnType.DATE;
-      } else if ("time".equals(def[1].toLowerCase())) {
-        type = ColumnType.TIME;
-      } else if ("long".equals(def[1].toLowerCase())) {
-        type = ColumnType.LONG;
-      } else {
+      try {
+        type = ColumnType.valueOf(def[1].toUpperCase(US));
+      } catch (IllegalArgumentException iae) {
         String errmsg = "Invalid UniqueKey type '" + def[1] + "'. Valid"
-            + " types are: int, string, timestamp, date, time, and long.";
+            + " types are: " + ColumnType.values();
         throw new InvalidConfigurationException(errmsg);
       }
-      if (tmpTypes.containsKey(name)) {
+      if (tmpNames.contains(name)) {
         String errmsg = "Invalid db.uniqueKey configuration: key name '"
             + name + "' was repeated.";
         throw new InvalidConfigurationException(errmsg);
+      } else if (tmpTypes.containsKey(name) && tmpTypes.get(name) != type) {
+        // The ukDecls contain two keys that differ only in case but are of
+        // different types. Force a case-sensitive type lookup by replacing
+        // the case-insensitive type map with a case-sensitive one.
```

Comment:
The "partial" and "copy" are important to me. I would change this line to: "the partial type map with a case-sensitive copy."